### ISSUE TITLE
Runtime issue with named export overrides

### ISSUE THREAD

**Issue by lexanth** [swc-project/swc#2101]
**Describe the bug**
Based on https://github.com/swc-project/swc/issues/1714

Valid JS is generated, but at runtime if the override is not the same (identity-wise) as the export from `*` the code won't load.

**Input code**
```
import { customRender } from './customRender';

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
```

**Config**

```json
{
  "sourceMaps": true,
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "automatic",
        "refresh": true
      },
      "hidden": {
        "jest": true
      }
    }
  },
  "module": {
    "type": "commonjs"
  },
  "env": {
    "coreJs": 3,
    "mode": "usage"
  }
}
```

**Expected behavior**
The transpilation should respect the behaviour that the named exports can override the wildcard exports.
e.g. [Babel REPL](https://babeljs.io/repl#?browsers=node%2014&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbzgYwK4GcYRAJQKYB2AJnlHAL5wBmU2cA5AHQD0am2-xp9A3AFB9mzOFDwBaPAA9IsOHgBupAJ4wAFsAIBzPlJnwAVNVogGAARh5MGzWIA2wAEZQAhlCXNRz5DF4ChcCEUoKGASEUISMhA8NQgiHWloeCQ2LFwI0jhndHCuMnJ-IA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env,react&prettier=false&targets=Node-14&version=7.14.1&externalPlugins=) generates the `_exportNames` object for overrides and then excludes those when defining the wildcard export.
```
var _exportNames = {
  render: true
};

Object.keys(_react).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return; // <-- This check is missing in the swc output
  if (key in exports && exports[key] === _react[key]) return; // <-- previous fix added this check, but only works when the override is the same as the * export
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _react[key];
    }
  });
});
```

**Version**
The version of @swc/core: 1.2.77

**Additional context**
This code should have this behaviour too:
```
export { customRender as render } from './customRender';

// re-export everything, but keep named 'render' exported above
export * from '@testing-library/react';
```

**Comment by th3fallen** [comment]
i'd like to second this im running into this issue too, is there any workaround?

**Comment by joshuacc** [comment]
Just wanted to add that I have also run into this issue. I have looked into fixing it, but haven't yet been able to spend enough time to figure out exactly where the fix needs to be made. The fact that I've barely touched Rust before certainly doesn't help. :)

**Comment by JeremyGrieshop** [comment]
Is this still an issue with the latest release?  I merged a fix in a few weeks ago to address something similar to this issue, but it's not 100% fixed yet in all cases, so I was planning a new PR for it.

**Comment by th3fallen** [comment]
So im not sure if this is relevant but I still get this issue when doing...

```import * as auth from 'shared/auth/auth';

auth.getUserId = jest.fn().mockReturnValue(1);
auth.getTokenInfo = jest.fn().mockReturnValue({ limited: false });

import { render } from '@testing-library/react';
//re-export everything
export * from '@testing-library/react';

//override render method
export { default as render } from 'tests/renderers/baseRenderer';
export { default as renderDialog } from 'tests/renderers/dialogRenderer';

export { default as selectEvent } from 'react-select-event';
export * from 'tests/helpers/reactSelect';
export { default as userEvent } from '@testing-library/user-event';
export { screen } from 'query-extensions';
```

the screen export is not available for anything importing from that file.

```TypeError: Cannot read properties of undefined (reading 'screen')



      at Object.get [as screen] (app/tests/testUtils.js:37:33)
      at app/tests/testUtils.js:46:34````

**Comment by JeremyGrieshop** [comment]
How did you verify this?  The PR was merged, but will not be released until 1.2.114 milestone.

> So im not sure if this is relevant but I still get this issue when doing...
> 
> ```
> 
> auth.getUserId = jest.fn().mockReturnValue(1);
> auth.getTokenInfo = jest.fn().mockReturnValue({ limited: false });
> 
> import { render } from '@testing-library/react';
> //re-export everything
> export * from '@testing-library/react';
> 
> //override render method
> export { default as render } from 'tests/renderers/baseRenderer';
> export { default as renderDialog } from 'tests/renderers/dialogRenderer';
> 
> export { default as selectEvent } from 'react-select-event';
> export * from 'tests/helpers/reactSelect';
> export { default as userEvent } from '@testing-library/user-event';
> export { screen } from 'query-extensions';
> ```
> 
> the screen export is not available for anything importing from that file.
> 
> ```
> 
> 
> 
>       at Object.get [as screen] (app/tests/testUtils.js:37:33)
>       at app/tests/testUtils.js:46:34````
> ```

**Comment by th3fallen** [comment]
ah derp i saw the release this morning and assumed. great catch
